src/aws/flb_aws_credentials_ec2.c

```diff
@@ -49,7 +48,10 @@ struct flb_aws_provider_ec2 {
     time_t next_refresh;
 
     /* upstream connection to IMDS */
-     struct flb_aws_client *client;
+    struct flb_aws_client *client;
+    
+    /* IMDS interface */
+    struct flb_aws_imds *imds_interface;
```

Comment:
Should I rename this to just **imds**? On second thought, `imds_interface` could be confusing since this is not a c++ style abstract class, but rather an interface in the webster sense: "a point where two systems, subjects, organizations, etc. meet and interact."

Reply:
honestly... I don't have an opinion on this... I think either are clear and readable